V3SampleApp/DetailViewController.swift

```diff
@@ -32,11 +36,21 @@ class DetailViewController: UIViewController {
             tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
         ])
         
+        controller.delegate = self
+        controller.startUpdating()
+        
         tableView.reloadData()
     }
 
 }
 
+extension DetailViewController: ChannelControllerDelegate {
+    func channelController<ExtraData: ExtraDataTypes>(_ channelController: ChannelController<ExtraData>, didUpdateChannel channel: ChannelModel<ExtraData>) {
+        // TODO is it possible to avoid force casting?
+        self.messages = channel.latestMessages as! [Message]
+    }
```

Comment:
This is an interesting problem. We have to make the delegate concrete over the extra data types the controller uses. I'm actually working on it as a part of CIS-199 but don't have a good solution yet :/ I say you don't need to worry about this as a part of this PR 👍

Reply:
At the end our ViewControllers will be with generic in anyway.